app/sessionStore.js

```diff
@@ -59,7 +59,14 @@ module.exports.saveAppState = (payload, isShutdown) => {
         wndPayload.tabs = wndPayload.tabs.filter((tab) => !tab.isPrivate)
       })
     } else {
-      delete payload.perWindowState
+      // Reset the window state except for the ui data
+      payload.perWindowState.forEach((wndPayload) => {
```

Comment:
specifically [] is not falsy so maybe the check in the block above is superfluous `if (payload.perWindowState && savePerWindowState) {` but it makes me wonder if it can be undefined and not empty sometimes. It could be an old check.